### ISSUE TITLE
UX Support for Bedrock users

### ISSUE THREAD

**Issue by data-dataninja** [elBukkit/MagicPlugin#768]
Any recommendation to configure for bedrock users? The closest example i can find is norp, but still great. As they dont have resource pack, nor can left/right click (only short press for left click/ long press for right click). Or if you can add some code to specifically support bedrock phones?

**Comment by NathanWolf** [comment]
Firstly I'd get on the 8.0 dev build track. I am nearing release, it should be pretty stable, and one of the main goals of it is to make the plugin easier to use without a resource pack (having Bedrock partially in mind).

https://jenkins.elmakers.com/job/MagicPlugin/3293/

I don't think they will be able to use a wand, though. The magic sword might be useable with some tweaks, but what I would recommend going for is the spellbook.

With this they can swing at the ground or hold the icon (to 'drop' it) to bring up their spell inventory. They can then take the spells and put them in their hotbar. From there tapping the spell will cast.

I have not yet tested this much, I'm not sure for instance if they can switch pages once they get more spells than fits an inventory. I'm also not sure the inventory sizes align.

In short, it's a work in progress but I am trying to get this working as best I can as well.

I have not yet looked into what, if any, plugin support there is for doing special things with Geyser connections.

**Comment by data-dataninja** [comment]
Thank you so much for the prompt reply! 
Will def test out the magic sword in v8.0.
I originally thought I just need to change the spell icons to a vanilla item 
Then i found out that on phone short/long press doesnt always convert left/right click (maybe its a geyser issue). 
Then i tested using controller on the phones works great (rt/lt to r/l click) - so not sure whats wrong with touch screen.
So i thought, i just 1. config the spell items to lets say(slimball) & wand to a stick 2. limit all spells to less than a hotbar & thats all i need for my server.
But i will def test out the magic sword in v8.0 & let u know!

**Comment by data-dataninja** [comment]
Just test out the v8.0. Its great I has all most all the things (hoe to a stick, no rp), & works perf out of box!
However just 1 thing, with bedrock, short/long press still dont work as left/right click, controller works perfectly thou (lt/rt - left/right click).
So then I went on trying to test the magic sword/ spellbook - couldnt find much info on to bring them up & stuck.
I guess the only thing is to get the touch screen work ? (short/long press as left/right click - or its a geyser issue?)
here is my server (still under dev pre-launch)
dgworlds.apexmc.co
i want to eventually run magic on a world of warcraft map (crossplay java/bedrock)
Let me know what i should do next!

**Comment by NathanWolf** [comment]
As far as I know the clicking not working is just how touchscreens work on Bedrock :(
You only swing if there is a mob in range you can hit, and you only "interact" when appropriate.
I did not even think of trying a controller, I'm glad that works but it's probably too much of an edge case for you to rely on.

Try `/mgive spellbook` and see how that works!  The sword is `/mgive magicsword` but the way the sword is set up you'd have to "interact" to open the inventory, so it wouldn't be too good out of the box. I think we could tweak that, but we'd have to make it so you only have one hotbar. This would do it, I think, in wands.yml
```
base_sword:
   drop: toggle
   hotbar_count: 1
```

That would make it so you could open the spell inventory with the drop action, which seems to be the only reliable interaction on touchscreens, even if it is a bit of a pain (very long press)

**Comment by data-dataninja** [comment]
AWESOME!
I tested both spellbook & magicsword, they all work great! (out of box)
Only one thing - on bedrock magicsword page flip doesnt work - (tried mutiple ways, works on spellbook).
So my concept is - (vanilia + magic - lite/simple)
ie:
# 1 magic item - likely spellbook
# player magicshop by (clicking npc)/ no enchantment table feature
# spells bound to player(not magic item - can be purchased/in a kit) 
# spells ranked to player groups & progression throu npc
# magic npc only teachskill /  givekit both cost money instead of sp (sp for upskill only)
# ? keep spells in 1 page 
# ? test block change spells (if to exclude, because of griefing/performance)

Let me know what you think, also wether we should make an example for this (maybe called vanila/lite?)

btw i got maps for ltor so its middle-earth map with magic - my server:
dgworlds.apexmc.co

THANKS!!!

**Comment by data-dataninja** [comment]
Hi Nathan,

Can I please get some help on configuring: 

1: spellbook not binding to player, spells bind to player
2: spells perm to player groups 
3. exclude block change spells / or to reduce the block change effect on those spells

THANKS!

**Comment by NathanWolf** [comment]
1. `/mconfig configure wand spellbook bound false`
2. Individual spell permissions can be controlled by negating `Magic.cast.<spell>`
3. If you grab the latest dev build, you can do `/mconfig example set safe`

**Comment by data-dataninja** [comment]
> 1. `/mconfig configure wand spellbook bound false`
works great thanks!
> 2. Individual spell permissions can be controlled by negating `Magic.cast.<spell>`
thanks will go this route 
> 3. If you grab the latest dev build, you can do `/mconfig example set safe`
doesnt work

/mconfig example command not available, i used the 8.0 latest build. also I tried to change the config example to safe, doest work as i thought theres an example called safe mode.

THANKS please help!

**Comment by NathanWolf** [comment]
Can you please share the output of `/magic` ?

**Comment by data-dataninja** [comment]
Magic 8.0-SNAPSHOT-c002aaf

/mconfig all options are: 
clean/configure/disable/enable

THANKS!

**Comment by data-dataninja** [comment]
Wait I have the reloaded the newest version & /mconfig example set safe works  awesome!!!THANKS
